main/modes/mode_gamepad.c

```diff
@@ -337,13 +687,31 @@ void gamepadButtonCb(buttonEvt_t* evt)
  */
 void gamepadTouchCb(touch_event_t* evt)
 {
-    if(evt->down)
-    {
-        gamepad->gpState.buttons |= touchMap[evt->pad];
-    }
-    else
-    {
-        gamepad->gpState.buttons &= ~touchMap[evt->pad];
+    switch(gamepad->gamepadType){
+        case GAMEPAD_GENERIC: {
+            if(evt->down)
+            {
+                gamepad->gpState.buttons |= touchMap[evt->pad];
+            }
+            else
+            {
+                gamepad->gpState.buttons &= ~touchMap[evt->pad];
+            }
+
+            break;
+        }
+        case GAMEPAD_NS: {
```

Comment:
The mapping here feels a little strange to me for two reasons.
1. The silkscreened X and Y aren't mapped to `GAMEPAD_BUTTON_X` and `GAMEPAD_BUTTON_Y`
2. Multiple pads are mapped to the same button. I'd rather have pads 0, 2, and 4 mapped to buttons and have 1 and 3 mapped to 0 rather than duplicated.
```C
GAMEPAD_BUTTON_TL,
GAMEPAD_BUTTON_TL,
GAMEPAD_BUTTON_Z,
GAMEPAD_BUTTON_TR,
GAMEPAD_BUTTON_TR,
```

Reply:
This is intentional, though I’m still thinking it through. It’s really difficult to press only 1 of 5 touch buttons, so I made it easier to press L and R (more games use those than X and Y).

Reply:
If you want to turn two pads into one pad, I think you need to logically OR them together. Otherwise releasing the second while the first is pressed will release the button, and that's not great.

That aside, people _will_ be confused to have something labeled X&Y actually be L&R. Taking a step back here, aside from being a neat gimmick, what specific games would you say are swadge gamepad compatible? What buttons do they require? You already mentioned Smash was not possible unless you map the D-Pad buttons to analog extremes rather than the D-Pad.

Reply:
Added Y and X to the extremities of the touch map.

Reply:
Is this actually better? Just testing in PC mode, I was able to consistently hit pads 0 and 4 without hitting an adjacent one, but I was not able to hit pads 1 or 3 without also getting an adjacent one. While I commend trying to jam all buttons on here, I think it'll leads to more misclicks than anything. Using only pads 0, 2, and 4 leave 1 and 3 as necessary buffer zones.

```C
const hid_gamepad_button_bm_t touchMapNs[] =
{
    GAMEPAD_BUTTON_Y,
    GAMEPAD_BUTTON_TL,
    GAMEPAD_BUTTON_Z,
    GAMEPAD_BUTTON_TR,
    GAMEPAD_BUTTON_X,
};
```

Reply:
It will result in misclicks outside of the intended uses (NES and Arcade games, which generally use L and R but not Y and X), but Y and X will still behave as a user would expect based on the PCB labels. Z doesn’t map to anything on the Switch so it’s a buffer zone. 

Thought this would be the best possible compromise.

Reply:
Well thought out and when you put it like that, good enough for me